File: src/main/java/com/example/venka/javaspringsparkdemo/transformer/MessageTransformerImpl.java
package com.example.venka.javaspringsparkdemo.transformer;

import com.example.venka.javaspringsparkdemo.model.Message;
import org.bson.Document;
import org.bson.types.ObjectId;
import org.springframework.stereotype.Component;

@Component
public class MessageTransformerImpl implements MessageTransformer {

    public Document toDocument(final Message message) {
        final Document document = new Document();
        document.append("text", message.getText());
        return document;
    }

    public Message fromDocument(final Document document) {
        final Message message = new Message();
        message.setId(document.getObjectId("_id"));
        message.setText(document.getString("text"));
        return message;
    }

    @Override
    public Document toDocument(ObjectId id) {
        final Document document = new Document();
        document.append("_id", id);
        return document;
    }
}

File: src/main/kotlin/com/example/venka/kotlinspringsparkdemo/transformer/MessageTransformerImpl.kt
package com.example.venka.kotlinspringsparkdemo.transformer

import com.example.venka.kotlinspringsparkdemo.model.Message
import org.bson.Document
import org.bson.types.ObjectId
import org.springframework.stereotype.Component

@Component
class MessageTransformerImpl : MessageTransformer {

    override fun toDocument(message: Message): Document {
        val document = Document()
        document.append("text", message.text)
        return document
    }

    override fun fromDocument(document: Document): Message {
        return Message(document.getObjectId("_id")!!, document.getString("text"))
    }

    override fun toDocument(id: ObjectId): Document {
        val document = Document()
        document.append("_id", id)
        return document
    }
}
